frontend/src/views/camp/CampProgram.vue

```diff
@@ -6,24 +6,57 @@ Show all activity schedule entries of a single period.
   <content-card :title="$tc('views.camp.picasso.title')" toolbar>
     <template #title-actions>
       <period-switcher v-if="$vuetify.breakpoint.xsOnly" :period="period" />
-
+      <v-spacer />
       <v-tooltip :disabled="isContributor" bottom>
-        <template #activator="{ on, attrs }">
-          <div
-            v-bind="attrs"
-            v-on="on">
-            <e-switch
-              v-model="editMode"
-              :disabled="!isContributor"
-              :label="$tc('views.camp.picasso.editMode')" />
-          </div>
+        <template #activator="{ on }">
+          <v-icon v-if="editMode"
+                  small color="grey"
+                  v-on="on">
+            mdi-lock-open
+          </v-icon>
+          <v-icon v-else
+                  small color="grey"
+                  v-on="on">
+            mdi-lock
+          </v-icon>
         </template>
         <span>{{ $tc('views.camp.picasso.guestsCannotEdit') }}</span>
       </v-tooltip>
-
-      <pdf-download-button-nuxt :config="printConfig()" class="ml-3" />
-
-      <pdf-download-button-react :config="printConfig()" class="ml-3" />
+      <v-menu offset-y>
+        <template #activator="{ on, attrs }">
+          <v-btn icon
+                 v-bind="attrs"
+                 v-on="on">
+            <v-icon>mdi-dots-horizontal</v-icon>
+          </v-btn>
+        </template>
+        <v-list class="py-0">
+          <v-list-item @click="editMode = !editMode">
+            <v-list-item-icon>
+              <v-icon v-if="editMode">mdi-lock</v-icon>
+              <v-icon v-else>mdi-lock-open</v-icon>
+            </v-list-item-icon>
+            <v-list-item-title>
+              {{ editMode ? 'Sperren' : 'Entsperren' }}
+            </v-list-item-title>
+          </v-list-item>
+          <v-divider />
+          <v-list-item @click="printNuxt">
```

Comment:
Du musst jetzt hier ziemlich viel print-Login in die `CampProgram.vue` Komponente ziehen. Mach doch eine Komponente analog zu `pdf-download-button-nuxt`, welche als v-list-item rendert anstatt als Button (z.B. pdf-download-menu-item-nuxt). Dort kannst du dir entweder download.js als Methode hineinziehen, oder nutzt ein Mixin.

`mdi-timer-sand` als statisches Icon finde ich nicht ideal, da gefällt mir der Spinner besser. Visuell einfach zu sehen, dass da was am Rendern ist.

Mit dem @click kann ich jetzt das PDF drucken mehrmals triggern (das ist bei VBtn automatisch disabled, wenn der loading-Spinner läuft).

Gibt es eine Möglichkeit, das Menu offen zu lassen, wenn ich auf print klicke?

Reply:
Fixed in d038380 and 394f03d

Reply:
Würdest du es nur offen lassen, wenn man auf die Downloads klickt?